homeassistant/components/netatmo/sensor.py

```diff
@@ -266,34 +276,29 @@ def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
-    @property
-    def unique_id(self):
-        """Return the unique ID for this sensor."""
-        return self._unique_id
-
     @property
     def available(self):
-        """Return True if entity is available."""
+        """Return entity availability."""
         return self._state is not None
 
-    def update(self):
-        """Get the latest data from Netatmo API and updates the states."""
-        self.netatmo_data.update()
-        if self.netatmo_data.data is None:
+    @callback
+    def async_update_callback(self):
+        """Update the entity's state."""
+        if self._data is None:
             if self._state is None:
                 return
             _LOGGER.warning("No data from update")
             self._state = None
             return
 
-        data = self.netatmo_data.data.get(self._module_id)
+        data = self._data.get_last_data(station_id=self._station_id, exclude=3600).get(
+            self._id
+        )
 
         if data is None:
             if self._state:
-                _LOGGER.debug(
-                    "No data found for %s (%s)", self.module_name, self._module_id
-                )
-                _LOGGER.debug("data: %s", self.netatmo_data.data)
+                _LOGGER.debug("No data found for %s (%s)", self._device_name, self._id)
+                _LOGGER.debug("data: %s", self._data)
```

Comment:
```suggestion
                _LOGGER.debug("No data (%s) found for %s (%s)", self._data, self._device_name, self._id)
```

I usually the ask is to combine logger statements that are next to each other.  I'm not sure what these actually output so suggestion may be offbase